docs/DeveloperGuide.md

```diff
@@ -115,12 +115,41 @@ The `JsonAdaptedCompany` also contains a list of roles in `List<JsonAdaptedRole>
 ---
 ## **Implementation**
 
+### Find feature
+
+The `find` feature allows users to filter the company list by specifying company name keywords and role name keywords.
+
+#### Implementation
+The `find` command is primarily implemented by `FindCommandParser` a class that extends `Parser`, and `FindCommand`, which is a class that extends `Command`. For each `find` command, a `Predicate<Company>` object and a `Predicate<Role>` object will be created. Both `Predicate` objects contain a `test` function to determine whether the given company or role matches the keywords provided in the user input. 
+
+* Upon a valid user's input using the `find` command, the `FindCommandParser#parse()`creates a `CompanyNameContainsKeywordsPredicate` which extends `Predicate<Company>`, and a `RoleNameContainsKeywordsPredicate` which extends `Predicate<Role>`.
+* The `FindCommandParser#parse()` then uses the `CompanyNameContainsKeywordsPredicate` object and the `RoleNameContainsKeywordsPredicate` object to instantiate the `FindCommand`.
+* Then invoking the `FindCommand#execute()` method will update the `model` using the `Model#updateFilteredRoleList()` method, displaying only companies and roles that match the user input.
+
+
+![UML diagram of the Find feature](images/FindDiagram.png)
+
+The following sequence diagram shows how the `find` command operation works with the user input `find c/meta r/software mobile`:
+
+![UML diagram of the Find feature](images/FindSequenceDiagram.png)
```

Comment:
Should we add this in for all features as well. I've added it in for this feature.

Reply:
Come to think of it, it might seem be a bit repetitive; the `Logic` portion UML also has this issue too. Maybe we can add a universal note at the top of the developer guide once everyone pushes their parts.